### PR TITLE
cylint cleanup in combinatorial polyhedra

### PR DESCRIPTION
**Pull request by fchapoton**
fixes all cython-lint warnings in combinatorial_polyhedron folder

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.

### CODE REVIEW

File: src/sage/geometry/polyhedron/combinatorial_polyhedron/base.pyx [hunk 1]
```diff
@@ -2793,7 +2790,6 @@ cdef class CombinatorialPolyhedron(SageObject):
             if 'dual' in kwds and dual == -1 and kwds['dual'] in (False, True):
                 dual = int(kwds['dual'])
 
-        cdef FaceIterator face_iter
         if dual == -1:
```
Comment (kwankyu) [inline src/sage/geometry/polyhedron/combinatorial_polyhedron/base.pyx:2793]: Then where is `FaceIterator` defined? `FaceIterator` is used below.

Comment (kwankyu) [reply src/sage/geometry/polyhedron/combinatorial_polyhedron/base.pyx:2793]: Sorry for noise. It's defined in `base.pxd`.

**Review by kwankyu — APPROVED** [review]
LGTM.

**Comment by vbraun** [comment]
merge conflict

**Comment by fchapoton** [comment]
after a simple rebase, waiting for the checks

**Pull request merged**